third_party/sphincsplus/repos.bzl

```diff
@@ -19,3 +20,96 @@ def sphincsplus_repos():
             "https://storage.googleapis.com/ot-crypto-test-vectors/sphincsplus/sphincsplus_shake256_128s_round3.zip ",
         ],
     )
+    bare_repository(
+        name = "sphincsplus_fips205_ipd",
+        url = "https://github.com/sphincs/sphincsplus/archive/129b72c80e122a22a61f71b5d2b042770890ccee.tar.gz",
+        strip_prefix = "sphincsplus-129b72c80e122a22a61f71b5d2b042770890ccee",
+        additional_files_content = {
+            "ref/BUILD": """exports_files(glob(["**"]))
```

Comment:
Due to the size of this injected BUILD file, I would like to make it a separate file, but I was unable to get the `additional_files` argument to `bare_repository` to work correctly. @cfrantz do you know how I have to format the path in order to get this to work?

Reply:
You should consider `http_repository` (or `//rules/repo.bzl%http_repository_or_local`) instead of `bare_repository`.

`bare_repository` was developed to create a bazel repo that is a symlinked hierarchy with a bunch of BUILD files injected into the hierarchy.  Specifically, `bare_repository` was developed to inject an entire hierarchy of BUILD files into tock and libtock-rs.

I think the sphincsplus reference implementation is sufficiently simple in terms of directory hierarchy that you can probably write an appropriate `cc_library` rule at the root of the and simply add `build_file = "..."` the the http_repository rule.

In the interest of not being obtuse and answering your question, the `additional_build_files` argument accepts a [string:string] dict mapping the desired foreign-repo file location to the local (opentitan) repo file location:
```
additional_files = {
  "ref/BUILD": "third_party/sphincsplus/BUILD.spx-ref.bazel",
  "ref/params/BUILD": "third_party/sphincsplus/BUILD.spx-params.bazel",
  ...
}
```

I don't think the existing users of `bare_repository` actually make use of `additional_files`, so its not exactly well tested.

Reply:
Edit: you might need to prefix the local paths with `//` and/or call the `Label` constructor:
```
"foo/BUILD": Label("//third_party/foo/BUILD.foo.bazel"),
```

Reply:
Thanks for the feedback. I knew the standard `http_archive` doesn't work for fetching specific git commits, but I didn't know that `http_repository` would work.

Reply:
> Edit: you might need to prefix the local paths with `//` and/or call the `Label` constructor:
> 
> ```
> "foo/BUILD": Label("//third_party/foo/BUILD.foo.bazel"),
> ```

I've already tried this. It complains about the path not being a string.

Reply:
Sorry, I typo-ed that.  It's `http_archive`.

You need to construct the correct github URL to fetch the exact commit you want.

Something like:
```
https://github.com/$user/$repo/archive/$git_hash.tar.gz
```

Reply:
I got this working with `http_archive_or_local`.

I think I had incorrectly remembered that we couldn't use `http_archive` with specific git commits. When we added wycheproof as a non-vendored dependency we explicitly avoided depending on a specific git commit because we wanted to keep up with the repo.